tools/genomespace/genomespace_importer.py

```diff
@@ -1,220 +1,265 @@
-# Dan Blankenberg
-
+import argparse
 import json
-import optparse
 import os
-import shutil
-import tempfile
+import sys
+import uuid
 
-from six.moves import http_cookiejar
-from six.moves.urllib.parse import parse_qs, unquote_plus, urlparse
-from six.moves.urllib.request import build_opener, HTTPCookieProcessor, Request, urlopen
+from genomespaceclient import GenomeSpaceClient
 
+import galaxy
 from galaxy.datatypes import sniff
 from galaxy.datatypes.registry import Registry
 
-GENOMESPACE_API_VERSION_STRING = "v1.0"
-GENOMESPACE_SERVER_URL_PROPERTIES = "https://dm.genomespace.org/config/%s/serverurl.properties" % ( GENOMESPACE_API_VERSION_STRING )
-DEFAULT_GENOMESPACE_TOOLNAME = 'Galaxy'
-FILENAME_VALID_CHARS = '.-()[]0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ '
-
-CHUNK_SIZE = 2**20  # 1mb
-
-DEFAULT_GALAXY_EXT = "data"
 
-# genomespace format identifier is the URL
-GENOMESPACE_FORMAT_IDENTIFIER_TO_GENOMESPACE_EXT = {}  # TODO: fix this so it is not a global variable
-# TODO: we should use a better way to set up this mapping
+# Mappings for known genomespace formats to galaxy formats
 GENOMESPACE_EXT_TO_GALAXY_EXT = {'rifles': 'rifles',
                                  'lifes': 'lifes',
                                  'cn': 'cn',
-                                 'GTF': 'gtf',
+                                 'gtf': 'gtf',
                                  'res': 'res',
                                  'xcn': 'xcn',
                                  'lowercasetxt': 'lowercasetxt',
                                  'bed': 'bed',
-                                 'CBS': 'cbs',
+                                 'cbs': 'cbs',
                                  'genomicatab': 'genomicatab',
                                  'gxp': 'gxp',
                                  'reversedtxt': 'reversedtxt',
                                  'nowhitespace': 'nowhitespace',
                                  'unknown': 'unknown',
                                  'txt': 'txt',
                                  'uppercasetxt': 'uppercasetxt',
-                                 'GISTIC': 'gistic',
-                                 'GFF': 'gff',
+                                 'gistic': 'gistic',
+                                 'gff': 'gff',
                                  'gmt': 'gmt',
-                                 'gct': 'gct'}
+                                 'gct': 'gct',
+                                 'fasta': 'fasta',
+                                 'fna': 'fasta',
+                                 'fa': 'fasta',
+                                 'faa': 'fasta',
+                                 'ffn': 'fasta',
+                                 'fastq': 'fastqsanger',
+                                 'fq': 'fastqsanger',
+                                 'fqz': 'fastqsanger'}
```

Comment:
Sorry, I don't know how I got the order wrong, so that solves the one part, but you should remove the extensions that are not problematic in galaxy IMO.

So what you're saying is that only `fasta.gz` will be uncompressed ? Why only fasta.gz ? Or is that a type and you meant fastq.gz ? Or is that because it is an example of a non-compressed datatype ? Normally compressed datatypes like fastq.gz should not be uncompressed by `handle_uploaded_dataset`.

Now the important thing is that if you respect the genomespace supplied type, which maps compressed fastq (`fqz`) to fastqsanger, you must uncompress the file. But I would be very good if that doesn't happen and you would not special-case fastqs.